barretenberg/cpp/src/barretenberg/commitment_schemes/gemini/gemini_impl.hpp

```diff
@@ -308,18 +279,57 @@ std::vector<typename GeminiProver_<Curve>::Claim> GeminiProver_<Curve>::compute_
         }
     }
 
-    std::vector<Claim> opening_claims;
-    opening_claims.reserve(num_variables + 1);
+    return { std::move(A_0_pos), std::move(A_0_neg) };
+};
 
-    // Compute first opening pair {r, A₀(r)}
-    Fr evaluation = fold_polynomials[0].evaluate(r_challenge);
-    opening_claims.emplace_back(Claim{ fold_polynomials[0], { r_challenge, evaluation } });
-    // Compute the remaining m opening pairs {−r^{2ˡ}, Aₗ(−r^{2ˡ})}, l = 0, ..., m-1.
-    for (size_t l = 0; l < num_variables; ++l) {
-        evaluation = fold_polynomials[l + 1].evaluate(-r_squares[l]);
-        opening_claims.emplace_back(Claim{ fold_polynomials[l + 1], { -r_squares[l], evaluation } });
+/**
+
+ *
+ * @param mle_opening_point u = (u₀,...,uₘ₋₁) is the MLE opening point
+ * @param fold_polynomials vector of polynomials whose first two elements are F(X) = ∑ⱼ ρʲfⱼ(X)
+ * and G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X), and the next d-1 elements are Fold_i, i = 1, ..., d-1.
+ * @param r_challenge univariate opening challenge
+ */
+
+/**
+ * @brief Computes/aggragates d+1 univariate polynomial opening claims of the form {polynomial, (challenge, evaluation)}
+ *
+ * @details The d+1 evaluations are A₀₊(r), A₀₋(-r), and Aₗ(−r^{2ˡ}) for l = 1, ..., d-1, where the Aₗ are the fold
+ * polynomials.
+ *
+ * @param A_0_pos A₀₊
+ * @param A_0_neg A₀₋
+ * @param fold_polynomials Aₗ, l = 1, ..., d-1
+ * @param r_challenge
+ * @return std::vector<typename GeminiProver_<Curve>::Claim> d+1 univariate opening claims
+ */
+template <typename Curve>
+std::vector<typename GeminiProver_<Curve>::Claim> GeminiProver_<Curve>::construct_univariate_opening_claims(
+    const size_t log_n,
+    Polynomial&& A_0_pos,
+    Polynomial&& A_0_neg,
+    std::vector<Polynomial>&& fold_polynomials,
+    const Fr& r_challenge)
+{
+    std::vector<Claim> claims;
+
+    // Compute evaluation of partially evaluated batch polynomial (positive) A₀₊(r)
+    Fr a_0_pos = A_0_pos.evaluate(r_challenge);
+    claims.emplace_back(Claim{ A_0_pos, { r_challenge, a_0_pos } });
```

Comment:
must be moved as well
```suggestion
    claims.emplace_back(Claim{ A_0_pos, { r_challenge, a_0_pos } });
```

Reply:
yes, thanks. careless!